### PR TITLE
update pyqt to 5.15

### PR DESCRIPTION
**Pull request by Bleuzen**

### CODE REVIEW

File: studio.kx.carla.json [hunk 1]
```diff
@@ -46,51 +46,158 @@
...
     "shared-modules/linux-audio/fluidsynth2.json",
     {
-      "name": "pyqt5",
-      "config-opts": [
-        "--disable-static",
-        "--enable-x11"
+      "name": "pyqt",
+      "buildsystem": "simple",
+      "build-commands": [
+        ". ${FLATPAK_BUILDER_BUILDDIR}/build"
```
Comment (hfiguiere) [inline studio.kx.carla.json:60]:
```suggestion
        "./build"
```

**Comment by hfiguiere** [comment]
> Don't know why it failed yet. This error message at the end of the log doesn't look really useful to me. @hfiguiere @tinywrkb @falkTX any idea?

I see:

> /usr/bin/python3: No module named PyQt5.pyrcc_main

Which seems to trigger a fatal build error.

**Comment by hfiguiere** [comment]
I think if you set the permission to executable and update the commit it's ok. or can call it with `sh`.

**Comment by Bleuzen** [comment]
> /usr/bin/python3: No module named PyQt5.pyrcc_main

Hm but what doesn't find PyQt5 while it is installed? Why is it exactly there?

I guess it could be pyrcc5.
Some lines above:

`/app/bin/pyrcc5 ../../resources/resources.qrc -o resources_rc.py`

and below:

`make[1]: *** [Makefile:111: resources_rc.py] Error 1`

seems that resource handling doesn't work.
Maybe `pyrcc5` doesn't find PyQt5 because in the flatpak it is installed in a different prefix compared to the default one.

I tried to get it to find our PyQt5 install in the flatpak by setting the `PYTHONPATH` env var, but it still fails.
Idk. anymore :/

**Comment by Bleuzen** [comment]
I think I got it to build now. Damn, this took me way too long to figure out.
Turns out that pyrcc is not fully built be default.

From docs:
```
--no-tools
    New in version 5.3.
    The pyuic5, pyrcc5 and pylupdate5 tools will not be built.
```

So i removed the `--no-tools` build option for PyQt5 and thought it builds pyrcc. But turns out it doesn't really.
Only thing it does is install a wrapper script for it, but not pyrcc itself. We have to explicitly build the native lib too with another build option, which I had to guess:
```
--enable=pyrcc
```

It is such a great documentation and build system ahhh tf.
Anyway, it builds now.

But don't get too excited yet. Because it doesn't run xD

**Comment by falkTX** [comment]
> ```
> 
> ```
> 
> 
>     
>   
> It is such a great documentation and build system ahhh tf.

ah I know the pain.. pyqt build system is weird, also keeps changing on a regular basis.

**Comment by Bleuzen** [comment]
bot, build studio.kx.carla

**Comment by Bleuzen** [comment]
@falkTX  Can you review the Qt modules?
They are defined in this file:
https://github.com/flathub/studio.kx.carla/blob/pyqt-update/pyqt-build

I'm not sure which of those Carla really uses. Thinking about removing for example `QtDBus` and `QtMultimedia`.

**Comment by Bleuzen** [comment]
Latest build works :)

**Comment by falkTX** [comment]
> @falkTX Can you review the Qt modules?

they look correct as of now, only QtOpenGL missing (though this one is optional, can be turned off if it is troublesome)

**Pull request merged**